pip/compat.py

```diff
@@ -36,22 +41,65 @@
         cache_from_source = None
 
 
-if sys.version_info >= (3,):
-    def console_to_str(s):
-        try:
-            return s.decode(sys.__stdout__.encoding)
-        except UnicodeDecodeError:
-            return s.decode('utf_8')
+if sys.version_info > (3, 4):
```

Comment:
This check will succeed for 3.4.x releases, which isn't what you want. `sys.version_info >= (3, 5)` will do the right thing.